src/content/studies/StudyView.tsx

```diff
@@ -0,0 +1,553 @@
+
+  const { loading: retrievingStudy } = useQuery<GetApprovedStudyResp, GetApprovedStudyInput>(
+    GET_APPROVED_STUDY,
+    {
+      variables: { _id },
+      skip: !_id || _id === "new",
+      onCompleted: (data) => setFormValues(data?.getApprovedStudy),
+      onError: (error) =>
+        navigate(manageStudiesPageUrl, {
+          state: { error: error?.message || "Unable to fetch study." },
+        }),
+      context: { clientName: "backend" },
+      fetchPolicy: "no-cache",
+    }
+  );
+
+  const [updateApprovedStudy] = useMutation<UpdateApprovedStudyResp, UpdateApprovedStudyInput>(
+    UPDATE_APPROVED_STUDY,
+    {
+      context: { clientName: "backend" },
+      fetchPolicy: "no-cache",
+    }
+  );
+
+  const [createApprovedStudy] = useMutation<CreateApprovedStudyResp, CreateApprovedStudyInput>(
+    CREATE_APPROVED_STUDY,
+    {
+      context: { clientName: "backend" },
+      fetchPolicy: "no-cache",
+    }
+  );
+
+  const handlePreSubmit = (data: FormInput) => {
+    if (!isValidORCID(data?.ORCID)) {
+      setError("Invalid ORCID format.");
+      return;
+    }
+    if (!data?.controlledAccess && !data?.openAccess) {
+      setError("Invalid Access Type. Please select at least one Access Type.");
+      return;
+    }
+
+    setError(null);
+    onSubmit(data);
+  };
+
+  const onSubmit = async (data: FormInput) => {
+    setSaving(true);
+
+    const variables: CreateApprovedStudyInput | UpdateApprovedStudyInput = {
+      ...data,
+      name: data.studyName,
+      acronym: data.studyAbbreviation,
+    };
+
+    if (_id === "new") {
+      const { data: d, errors } = await createApprovedStudy({ variables }).catch((e) => ({
+        errors: e?.message,
+        data: null,
+      }));
+      setSaving(false);
+
+      if (errors || !d?.createApprovedStudy?._id) {
+        setError(errors || "Unable to create approved study.");
+        return;
+      }
+      enqueueSnackbar("This study has been successfully added.", {
+        variant: "default",
+      });
+    } else {
+      const { data: d, errors } = await updateApprovedStudy({
+        variables: { studyID: _id, ...variables },
+      }).catch((e) => ({ errors: e?.message, data: null }));
+      setSaving(false);
+
+      if (errors || !d?.updateApprovedStudy) {
+        setError(errors || "Unable to save changes");
+        return;
+      }
+
+      enqueueSnackbar("All changes have been saved.", { variant: "default" });
+      setFormValues(d.updateApprovedStudy);
+    }
+
+    setError(null);
+    navigate(manageStudiesPageUrl);
+  };
+
+  /**
+   * Updates the default form values after save or initial fetch
+   *
+   * @param data FormInput
+   */
+  const setFormValues = (data: FormInput, fields = editableFields) => {
+    const resetData = {};
+
+    fields.forEach((field) => {
+      if (data?.[field] === null) {
+        return;
+      }
+      resetData[field] = cloneDeep(data[field]);
+    });
+
+    reset(resetData);
+  };
```

Comment:
Instead of using `setFormValues`, we can just use `reset({ ...data })` directly. e.g. `setFormValues({ ...d.updateApprovedStudy })`.

I originally used this on the manage users page since the fieldset is different (e.g. Admins can change role/org/..., but Users can only edit their name). It somewhat got carried over to other places by accident.

```suggestion

```

Reply:
Thanks for the info, I have updated the implementation in [8400ce1](https://github.com/CBIIT/crdc-datahub-ui/pull/469/commits/8400ce119190db83331c81820215fffb03a702cb) where I instead map out the properties manually. The reason is that spreading out the result was also grabbing the `__typename` property and adding it into the form. LMK if you want me to approach this differently.